### PR TITLE
chore: update conformance test and example with encryption.Marshaler

### PR DESCRIPTION
**Pull request by DmitriyMV**
Use encryption.Marshaler in example and conformance test. Also update example to run on each go test invocation.

Signed-off-by: Dmitriy Matrenichev <dmitry.matrenichev@siderolabs.com>

### CODE REVIEW

**Review by frezbo — APPROVED** [review]

File: pkg/state/impl/store/bolt/conformance_test.go [hunk 1]
```diff
@@ -20,16 +20,24 @@ import (
...
 	t.Parallel()
 
 	tmpDir := t.TempDir()
 
-	marshaler := store.ProtobufMarshaler{}
+	marshaler := encryption.NewMarshaler(
+		store.ProtobufMarshaler{},
+		encryption.NewCipher(
+			encryption.KeyProviderFunc(func() ([]byte, error) {
+				return []byte("the god key to encrypt data with"), nil
```
[the remark this answers is not in the document]
Comment (DmitriyMV) [reply pkg/state/impl/store/bolt/conformance_test.go:35]: The funny part here is that this string is exactly 32-bytes. No religion intentions or anything, just a string I thought could be funny. Let me rethink it.

[the remark this answers is not in the document]
Comment (DmitriyMV) [reply pkg/state/impl/store/bolt/conformance_test.go]: Updated!

[the remark this answers is not in the document]
Comment (frezbo) [reply pkg/state/impl/store/bolt/conformance_test.go]: > The funny part here is that this string is exactly 32-bytes.

i didn't realize that easter egg :wink:

**Pull request merged**